### ISSUE TITLE
A specific package is always reinstalled since pacaur v4.7.0

### ISSUE THREAD

**Issue by pesho** [rmarquis/pacaur#640]
###### Version

```
pacaur 4.7.1
```

###### Description

After pacaur was updated from v4.6.10 to v4.7.0 the package `wire-desktop-bin` is always reinstalled, even though it hasn't actually changed. I have other AUR packages installed as well, but only this one is affected. Pacaur versions up to 4.6.10 didn't exhibit this.

###### Output

```
$ pacaur -Syua 
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
:: Starting AUR upgrade...
:: resolving dependencies...
:: looking for inter-conflicts...
:: wire-desktop-bin-2.11.2703-1 is up to date -- reinstalling

AUR Packages  (1)     Old Version  New Version

aur/wire-desktop-bin  2.11.2703-1  2.11.2703-1               

:: Proceed with installation? [Y/n]
```

###### Debug output

`bash -x pacaur -Syua > pacaur_debug.txt 2>&1`

https://gist.github.com/pesho/7b3fd9ca52f034894fe02329fdfbd289

**Comment by pesho** [comment]
Adding `--needed` seems to be a workaround for now:

```
$ pacaur -Syua --needed 
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
:: Starting AUR upgrade...
:: resolving dependencies...
:: looking for inter-conflicts...
:: wire-desktop-bin-2.11.2703-1 is up to date -- skipping
 there is nothing to do
```

**Comment by rmarquis** [comment]
I cannot reproduce, but it looks like something similar to #563 is happening.
Haven't found any issue yet, but I'll have a closer look tomorrow.

**Comment by rmarquis** [comment]
This is due to a sort issue, which is locale dependent.

```
+ for i in "${!allaurpkgs[@]}"
++ vercmp 2.11.2703-1 20161022-1
+ [[ -1 -gt 0 ]]
+ for i in "${!allaurpkgs[@]}"
++ vercmp 20161022-1 2.11.2703-1
+ [[ 1 -gt 0 ]]
```

Version comparison between local and AUR version of the `w_scan` and `wire-desktop-bin` are swapped.

**Comment by rmarquis** [comment]
Should be fixed in 10b882eb4e52b12c66e33a27e99cdebbaf7ed874. Feedback welcome.

**Comment by pesho** [comment]
> Should be fixed in 10b882e. Feedback welcome.

Awesome! Just tested the last commit, the issue is indeed fixed. Thanks a lot! :trophy:
